### PR TITLE
Some forme stuff (time display, icons)

### PR DESCRIPTION
**Pull request by casaper**

### CODE REVIEW

**Comment by huerlisi** [comment]
Couple remarks:

All in all I like the idea of introducing icons. But if we do we need to add them in almost all places, otherwise it will be inconsistent. This can be tricky as some field names do not have Icons in the default sets.

You used icons from the awesome fonts. But we're using the materialized icon set in other places. Can you try if you find materialized icons? See https://design.google.com/icons/

File: app/views/events/_form.html.haml [hunk 1]
```diff
@@ -1,7 +1,9 @@
 = simple_form_for resource do |f|
   .section
-    = render 'form_base', f: f
```
Comment (huerlisi) [inline app/views/events/_form.html.haml]: Why is the introduction of this case with the form needed?

**Comment by huerlisi** [comment]
We use git rebase to update feature branches with updates in branches instead of merges. Please do another rebase before merging.

File: app/views/events/_form.html.haml (hunk 1 above)
Comment (casaper) [inline app/views/events/_form.html.haml]: I did that cos the ```resource.start_at.strftime("%Y-%d-%m")``` won't work if resource is = ```nil```

**Comment by casaper** [comment]
@huerlisi #21
# Icon font:
I Chose font-awesome because for the calendar button in timeline fa icons where used.
Can be switched to material icons tho, no proplem with that ...

# icons everywhere
Actually I thought my css selectors for the icons to be quite specific – as Far as I noticed till now they're all at places I intended them to be. But possibly I intended too much ;)

**Comment by huerlisi** [comment]
Yeah, using materialized everywhere would be great, also in the timeline:-) Then we can get rid of the awesome stuff;-)

**Comment by huerlisi** [comment]
Regarding icons everywhere: I think the icons show up where they should. What I mean is that almost every field in almost every form should have icons assigned.

File: app/views/events/_form.html.haml (hunk 1 above)
Comment (huerlisi) [inline app/views/events/_form.html.haml]: Okay, I'll take a look. There are simpler ways to achieve that. The way you did it introduces quite some code duplication :sunglasses:

**Comment by casaper** [comment]
> almost every field in almost every form should have icons assigned

True that, but with some fields I was stretching my phantasy brain cells, but could not find a propper icon in font awesome. Maybe it'll be better with material icons ;)

**Comment by huerlisi** [comment]
Yeah that is exactly the issues with introducing such icons. I know that some designers shy away from icons for that reason;-)

File: app/views/events/_form.html.haml (hunk 1 above)
Comment (huerlisi) [inline app/views/events/_form.html.haml]: The following seems to do the trick:

`= f.time_field :start_at, value: f.object.start_at&.to_s(:time), class: 'timepicker'`

* better use `f.object` where `f` is the current form variable. This reduces the dependency on inherited resources etc. and uses the object that has been used to start the form
* The `&.` syntax does only call the following method if the object is not `nil`
* the `to_formatted_s` is aliased to `to_s`

I also think the `value` assignment is not needed for the date part.

Comment (casaper) [inline app/views/events/_form.html.haml]: Oh, phantastic. Didn't know abt ```&.````. This is awesome stuff. 👍
And I could have used the ```f```-var, somehow this was a leftover from me mangling things with byebug or so...
Thx for the lessons :)

Comment (casaper) [inline app/views/events/_form.html.haml]: Oh, phantastic. Didn't know abt ```&.````. This is awesome stuff. 👍
And I could have used the ```f```-var, somehow this was a leftover from me mangling things with byebug or so...
Thx for the lessons :)

Comment (casaper) [inline app/views/events/_form.html.haml]: Oh, phantastic. Didn't know abt ```&.````. This is awesome stuff. 👍
And I could have used the ```f```-var, somehow this was a leftover from me mangling things with byebug or so...
Thx for the lessons :)

File: app/views/events/timeline.html.haml [hunk 2]
```diff
@@ -29,27 +29,23 @@
...
+              %span.time
+                = event.end_at.to_s(:time)
+              %span.date
+                = event.end_at.strftime("%d.%m")
           .card-action
             - if can?(:edit, event) && event.persisted?
               %a{ href: url_for([:edit, event])} Edit event
             .avatars
               - event.attendances.each_with_index do |a, i|
-                = image_tag a.guest.avatar_url, class: ['circle', "attendance-state-#{a.state}"], style: "right: #{i*40}px"
```
Comment (huerlisi) [inline app/views/events/timeline.html.haml]: Why did you drop the `attendance-state-` class? Merge error?

**Comment by huerlisi** [comment]
Could you please rebase/squash the changes. It's quite a bunch of followup commits. You may use interactive rebase using `git rebase -i origin/master`. Or `git reset origin/master` and create fresh commits with `git add -p`.

I think you tackeled two topics: icons and time formats, so two commit would be nice.

And you should follow the commit convention;-)

**Comment by huerlisi** [comment]
Adding a small feature spec that does a round of the following steps would be great;-) You could check for the proper date format in there...

* open new form
* set some values, but not all
* save
* check for validation errors
* set the missing values
* save
* check for proper redirection

**Comment by huerlisi** [comment]
Did you close this PR on purpose? If you did rebase etc, you can still `git push --force` to keep this PR updated. This way we keep the discussion history.

**Comment by casaper** [comment]
I created a new branch with no merge and so, so I plan to make a new pull req from that branch

**Comment by huerlisi** [comment]
That's fine, too;-)

**Pull request closed without merging**